### DISCUSSION TITLE
Running bulk predictions locally using 100% CPU

### DISCUSSION THREAD

**Discussion by ishiland** [azavea/raster-vision#1693] [Q&A]
Is there a working example or suggested approach to running predictions locally using 100% CPU? I am only able to achieve 40-50% cpu usage when running predictions across several images. My script is looping over an array of input image paths and executing the prediction one at a time like this:
`predictor.predict([in_image_path], output_predict_path)`

The pattern I am using is similar as in [this issue](https://github.com/azavea/raster-vision/issues/1052).

**Comment by AdeelH** [comment]
Just to be clear, you are not using a GPU at all, right? Also, I'm assuming these are relatively large GeoTIFFs and that we are talking about the utilization during prediction on a single such image.

I don't know if CPU utilization scales the same way as GPU utilization, but you might try increasing the batch size.

**Reply by ishiland** [reply]
> Just to be clear, you are not using a GPU at all, right? Also, I'm assuming these are relatively large GeoTIFFs and that we are talking about the utilization during prediction on a single such image.

Yes this is correct. I have tried multiprocessing my for-loop but run into pickling issues. I will attempt to increase the batch size, thanks.

**Reply by AdeelH** [reply]
I wonder if running two instances of the same script in parallel might work. You might also want to try increasing `num_workers`.

Also, see also these tutorials for making predictions in a way that is more flexible than using the `Predictor` class: [one](https://docs.rastervision.io/en/0.20/usage/tutorials/lightning_workflow.html#Make-predictions-for-scene), [two](https://docs.rastervision.io/en/0.20/usage/tutorials/pred_and_eval_ss.html).
